bionic/flow.py

```diff
@@ -1526,28 +1524,50 @@ def reloading(self):
 
         self_name = self.name
 
-        module_names = set()
+        # Find the module that contains the flow.
+        candidate_flow_modules = set()
         for provider in state.providers_by_name.values():
             source_func = provider.get_source_func()
             if source_func is None:
                 continue
-            module_names.add(source_func.__module__)
-
-        blessed_candidate_flows = []
-        unblessed_candidate_flows = []
-        for module_name in module_names:
-            module = reload(module_registry[module_name])
+            module = module_registry[source_func.__module__]
             for key in dir(module):
                 element = getattr(module, key)
```

Comment:
We have this logic in two different places now. Would it make sense to factor it out into a `get_flow_from_module` function?

Reply:
Refactored into `_get_flows_from_module`